xdsl/interpreter.py

```diff
@@ -605,6 +605,11 @@ def run_op(self, op: Operation | str, inputs: PythonValues) -> PythonValues:
         if isinstance(op, str):
             op = self.get_op_for_symbol(op)
 
+        if (operands_count := len(op.operands)) != (inputs_count := len(inputs)):
+            raise InterpretationError(
+                f"Number of operands ({operands_count}) doesn't match the number of inputs ({inputs_count})."
+            )
+
```

Comment:
please move to _run_op instead, and also check number of results

Reply:
> please move to _run_op instead, and also check number of results

this causes a handful of tests to fail, im thinking i should just raise errors rn

Reply:
Let's fix the tests, that's why I asked you to open this PR :)

Reply:
yh, im not to sure how to fix these tests, 
the tests are:
`test_constant in test_arith`
`test_apply and test_functions in test_affine`

Reply:
The code in interpreter.py still seems broken

Reply:
the test needs to check that the number of results on the op is the same as the number of results returned by the impl

Reply:
okay, it now works i believe, i just have conflicts, you updated the way shapedarray performs